plugins/inputs/t128_metrics/t128_metrics_test.go

```diff
@@ -469,8 +469,15 @@ func TestT128MetricsRequestLimiting(t *testing.T) {
 }
 
 func TestTimoutUsedForRequests(t *testing.T) {
+	done := make(chan struct{}, 1)
+
 	fakeServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// hang request
+		select {
+		case <-done:
+		case <-time.After(10 * time.Second):
```

Comment:
This could probably well less than 10 s considering your timeout is 1 ms. It works though

Reply:
The `done` channel should always signal and cause this to exit. This 10 second delay can only happen in a failure case where the plugin timeout is not being honored.

Reply:
Oh yea, you're right. Should you remove it then? The test will fail if the configured plugin timeout is not being honored?

Reply:
There needs to be a timeout outside the expected client timeout or, in the failure case, the test will never complete.

Reply:
Ahh ok. Got it now